engine/src/java/com/runrev/android/Engine.java

```diff
@@ -1859,23 +1864,68 @@ else if (t_caminfo.facing == CameraCompat.CameraInfo.CAMERA_FACING_FRONT)
 		return new String(t_directions);
 	}
 
-	public void showPhotoPicker(String p_source, int p_width, int p_height)
-	{
-		m_photo_width = p_width;
-		m_photo_height = p_height;
-
-		if (p_source.equals("camera"))
-			showCamera();
-		else if (p_source.equals("album"))
-			showLibrary();
-		else if (p_source.equals("library"))
-			showLibrary();
-		else
-		{
-			doPhotoPickerError("source not available");
-		}
-	}
-
+    public static final int CAMERA_PERMISSION_REQUEST_CODE = 1;
```

Comment:
If we aren't actually using the request code then we could just call all requests with `1`. If we do need dispatching waits (see the other comment) then we probably need a unique code to come from the engine and map to the actual request.